### PR TITLE
Prevent Queue.qsize() from raising Error on Mac OS X

### PR DESCRIPTION
**Pull request by stegut01**
related issue: https://github.com/ARMmbed/greentea/issues/90

### CODE REVIEW

**Comment by PrzemekWirkus** [comment]
Looks nice. I will check this is a moment.

**Comment by PrzemekWirkus** [comment]
```
+---------------+---------------+---------------------------------+--------+--------------------+-------------+
| target        | platform_name | test suite                      | result | elapsed_time (sec) | copy_method |
+---------------+---------------+---------------------------------+--------+--------------------+-------------+
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | FAIL   | 12.34              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-dev_null      | OK     | 11.28              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-echo          | FAIL   | 19.86              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-generic_tests | OK     | 10.7               | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-rtc           | OK     | 20.83              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-stl_features  | OK     | 11.56              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-ticker        | OK     | 21.02              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-ticker_2      | OK     | 20.99              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-ticker_3      | OK     | 21.08              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-timeout       | OK     | 21.25              | shell       |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-wait_us       | OK     | 19.99              | shell       |
+---------------+---------------+---------------------------------+--------+--------------------+-------------+
mbedgt: test suite results: 2 FAIL / 9 OK
mbedgt: test case report:
+---------------+---------------+---------------------------------+-------------------------------------+--------+--------+--------+--------------------+
| target        | platform_name | test suite                      | test case                           | passed | failed | result | elapsed_time (sec) |
+---------------+---------------+---------------------------------+-------------------------------------+--------+--------+--------+--------------------+
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: %e %E float formatting   | 1      | 0      | OK     | 0.07               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: %f %f float formatting   | 0      | 1      | FAIL   | 0.3                |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: %g %g float formatting   | 1      | 0      | OK     | 0.06               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: %i %d integer formatting | 1      | 0      | OK     | 0.07               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: %u %d integer formatting | 1      | 0      | OK     | 0.07               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: %x %E integer formatting | 1      | 0      | OK     | 0.06               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: strpbrk                  | 1      | 0      | OK     | 0.05               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-c_strings     | C strings: strtok                   | 1      | 0      | OK     | 0.05               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-dev_null      | dev_null                            | 1      | 0      | OK     | 11.28              |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-echo          | Echo server: x16                    | 1      | 0      | OK     | 1.6                |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-echo          | Echo server: x32                    | 0      | 0      | ERROR  | 0.0                |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-generic_tests | Basic                               | 1      | 0      | OK     | 0.03               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-generic_tests | Blinky                              | 1      | 0      | OK     | 0.03               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-generic_tests | C++ heap                            | 1      | 0      | OK     | 0.09               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-generic_tests | C++ stack                           | 1      | 0      | OK     | 0.15               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-rtc           | RTC strftime                        | 1      | 0      | OK     | 10.43              |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-stl_features  | STL std::equal                      | 1      | 0      | OK     | 0.05               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-stl_features  | STL std::sort abs                   | 1      | 0      | OK     | 0.04               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-stl_features  | STL std::sort greater               | 1      | 0      | OK     | 0.05               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-stl_features  | STL std::transform                  | 1      | 0      | OK     | 0.05               |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-ticker        | Timers: 2 x tickers                 | 1      | 0      | OK     | 11.03              |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-ticker_2      | Timers: 1x ticker                   | 1      | 0      | OK     | 11.04              |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-ticker_3      | Timers: 2x callbacks                | 1      | 0      | OK     | 11.04              |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-timeout       | Timers: toggle on/off               | 1      | 0      | OK     | 11.25              |
| frdm-k64f-gcc | K64F          | mbed-drivers-test-wait_us       | Timers: wait_us                     | 1      | 0      | OK     | 10.06              |
+---------------+---------------+---------------------------------+-------------------------------------+--------+--------+--------+--------------------+
mbedgt: test case results: 1 FAIL / 23 OK / 1 ERROR
mbedgt: completed in 191.63 sec
mbedgt: exited with code 2
```

**Comment by PrzemekWirkus** [comment]
+1

**Comment by 0xc0170** [comment]
For future: Please read: https://docs.mbed.com/docs/getting-started-mbed-os/en/latest/Full_Guide/Code_Style/ , the section about the commit message.

File: mbed_host_tests/host_tests_runner/host_test_default.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def callback__notify_prn(key, value, timestamp):
             consume_preamble_events = True
             while (time() - start_time) < timeout_duration:
                 # Handle default events like timeout, host_test_name, ...
-                if event_queue.qsize():
+                if not event_queue.empty():
```
Comment (mazimkhan) [inline mbed_host_tests/host_tests_runner/host_test_default.py]: While ```not event_queue.empty():``` yields False we would be in a tight loop in ```while``` on line #118. 
This if statement is not required as execution of this loop depends on data coming in the ```event_queue```.

Implementation can look like:
```
while True:
    rem = timeout_duration - (time() - start_time)
    if rem > 0: 
        break

    try:
        (key, value, timestamp) = event_queue.get(timeout=rem)
    except QueueEmpty:
        continue
```

**Pull request merged**